### PR TITLE
Salesforce Refresh Token Detector

### LINKED ISSUE #4182: New Detector for Salesforce
## Description

Salesforce OAuth2 credentials including the consumer and secret key pair are used to authenticate and authorize applications that need access to Salesforce data. These credentials are part of a connected app, which allows external services to securely interact with Salesforce APIs. The consumer key acts as a unique identifier for the app, while the consumer secret is used to verify the app’s authenticity when requesting an access token. Additionally, refresh tokens play a crucial role in maintaining access without requiring users to log in repeatedly. When an access token expires, a refresh token allows the application to request a new access token without user intervention. This ensures seamless and uninterrupted access to Salesforce resources while enhancing security by limiting the lifespan of access tokens. The current Salesforce detector is only scanning for access tokens that have a short TTL by default, but it is not scanning for either the long live consumer/secret key pair or the refresh token that are utilized to generate the access token in the first place. The two secret types should be able to be mostly identified by the regex patterns below.

Refresh Token: 5aep861[a-zA-Z0-9._=]+

Consumer Key: 3MVG9[a-zA-Z0-9+/=_.]+ 
Secret Key: [A-Za-z0-9+/=_.]{64} OR [0-9]{19}

### Preferred Solution

Expand the current Salesforce detector to identify the three credential types that make up the Salesforce OAuth2 workflow. The provided patterns should make for a decent baseline on the expected match.

### PR DESCRIPTION
**Pull request by shahzadhaider1**
### Description:
This PR adds a new detector for Salesforce OAuth 2.0 refresh tokens. Refresh tokens are long-lived credentials that allow applications to obtain new access tokens without requiring user interaction, making them critical security assets that need to be detected in code repositories.

## What's Added
- **New Detector**: `salesforcerefreshtoken` package with full detection and verification capabilities
- **Pattern Recognition**: Detects Salesforce refresh tokens (starting with `5AEP861`), consumer keys (starting with `3MVG9`), and consumer secrets
- **Verification**: Validates refresh tokens using Salesforce's OAuth 2.0 introspection endpoint
- **Comprehensive Tests**: Unit tests for pattern matching and integration tests for verification

### Detection Patterns
- **Refresh Token**: `(?i)\b(5AEP861[a-zA-Z0-9._=]{80,})\b`
- **Consumer Key**: `\b(3MVG9[0-9a-zA-Z._+/=]{80,251})`
- **Consumer Secret**: Uses `PrefixRegex` with keywords `["salesforce", "consumer", "secret"]` + `\b([A-Za-z0-9+/=.]{64}|[0-9]{19})\b`

### Verification Strategy
The detector uses Salesforce's OAuth 2.0 introspection endpoint (`/services/oauth2/introspect`) to validate refresh tokens:
- **200 OK**: Parses JSON response to check if token is `active`
- **400 Bad Request**: Invalid token format or expired token
- **401 Unauthorized**: Invalid client credentials
- **Other**: Unexpected errors are properly handled

### Triple Component Requirement
The detector requires all three components (refresh token, consumer key, consumer secret) to be present:
- Creates a cartesian product of all found combinations
- Returns early if any component is missing (performance optimization)
- Handles multiple instances of each component type

### Pattern Tests
- ✅ Simple case: one of each component
- ✅ Combinatorial cases: multiple keys/tokens/secrets
- ✅ Negative cases: invalid formats, missing components
- ✅ Edge cases: boundary conditions for pattern matching

### Integration Tests
- ✅ Valid credentials (verified)
- ✅ Invalid credentials (unverified)
- ✅ Multiple findings with mixed validity
- ✅ Missing components (no results)
- ✅ Timeout handling
- ✅ Unexpected API responses

## Keywords
The detector uses the following keywords for efficient pre-filtering:
- `salesforce`
- `5AEP861`
- `3MVG9`

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/welcome/install/#local-installation))?

salesforce-refresh-token-detector-tests

### CODE REVIEW

File: pkg/detectors/salesforcerefreshtoken/salesforcerefreshtoken.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+			return false, fmt.Errorf("failed to read error response body: %w", err)
+		}
+		var apiResp introspectAPIResponse
+		if err := json.Unmarshal(bodyBytes, &apiResp); err != nil {
+			return false, fmt.Errorf("failed to unmarshal error response: %w (body: %s)", err, string(bodyBytes))
+		}
+
+		if !apiResp.Active {
+			return false, nil
+		}
```
Comment (abmussani) [inline pkg/detectors/salesforcerefreshtoken/salesforcerefreshtoken.go:134]: Does not it show that credentials are valid but the token is not active ?

**Review by abmussani — COMMENTED** [review]
a minor comment

File: pkg/detectors/salesforcerefreshtoken/salesforcerefreshtoken.go (hunk 1 above)
Comment (shahzadhaider1) [reply pkg/detectors/salesforcerefreshtoken/salesforcerefreshtoken.go:134]: Yes, it means that the credentials are valid, that is why we were able to get the status of the refresh token.

**Review by kashifkhan0771 — APPROVED** [review]

File: pkg/detectors/salesforcerefreshtoken/salesforcerefreshtoken.go (hunk 1 above)
Comment (shahzadhaider1) [reply pkg/detectors/salesforcerefreshtoken/salesforcerefreshtoken.go:134]: Just to add - the validity of the credentials themselves is handled by a separate detector we already have, [salesforceoauth2](https://github.com/trufflesecurity/trufflehog/blob/main/pkg/detectors/salesforceoauth2/salesforceoauth2.go). This particular detector is specifically designed to identify and verify refresh tokens, which is why the focus here is limited to that aspect.
